dot/network/message.go

```diff
@@ -354,3 +360,80 @@ func (cm *ConsensusMessage) Hash() (common.Hash, error) {
 	}
 	return common.Blake2bHash(encMsg)
 }
+
+// NewSingleBlockRequestMessage returns a request to retrieve a single block
+func NewSingleBlockRequestMessage(blockHash common.Hash, requestedData byte) *BlockRequestMessage {
+	one := uint32(1)
+	return &BlockRequestMessage{
+		RequestedData: requestedData,
+		StartingBlock: *variadic.MustNewUint32OrHash(blockHash),
+		Direction:     Descending,
+		Max:           &one,
+	}
+}
+
+// NewDescendingBlockRequest returns a descending block request message
+func NewDescendingBlockRequest(blockHash common.Hash, amount uint32, requestedData byte) *BlockRequestMessage {
+	return &BlockRequestMessage{
+		RequestedData: requestedData,
+		StartingBlock: *variadic.MustNewUint32OrHash(blockHash),
+		Direction:     Descending,
+		Max:           &amount,
+	}
+}
+
+func NewAscedingBlockRequests(startNumber, targetNumber uint, requestedData byte) []*BlockRequestMessage {
+	if startNumber > targetNumber {
+		return []*BlockRequestMessage{}
+	}
+
+	diff := targetNumber - startNumber
+	fmt.Printf("target number: %d, start number: %d = %d\n", targetNumber, startNumber, diff)
+
+	// start and end block are the same, just request 1 block
+	if diff == 0 {
+		one := uint32(1)
+		return []*BlockRequestMessage{
+			{
+				RequestedData: requestedData,
+				StartingBlock: *variadic.MustNewUint32OrHash(uint32(startNumber)),
+				Direction:     Ascending,
+				Max:           &one,
+			},
```

Comment:
could just use a single function say `NewBlockRequest` instead of `NewDescendingBlockRequest` and `NewSingleBlockRequestMessage` with an extra direction parameter and reuse it here.